test/e2e/setup/setup.go

```diff
@@ -164,10 +169,33 @@ func deployOperator() error {
 	); err != nil {
 		return errors.Errorf("error building operator deployment: %s", err)
 	}
+
+	if err := wait.PollImmediate(time.Second, 30*time.Second, hasDeploymentRequiredReplicas(dep)); err != nil {
+		return errors.New("error building operator deployment: the deployment does not have the required replicas")
+	}
+
 	fmt.Println("Successfully created the operator Deployment")
 	return nil
 }
 
+// hasDeploymentRequiredReplicas returns a condition function that indicates whether the given deployment
+// currently has the required amount of replicas in the ready state as specified in spec.replicas
+func hasDeploymentRequiredReplicas(dep *appsv1.Deployment) wait.ConditionFunc {
+	return func() (bool, error) {
+		err := e2eutil.TestClient.Get(context.TODO(),
+			types.NamespacedName{Name: dep.Name,
+				Namespace: e2eutil.OperatorNamespace},
+			dep)
+		if err != nil {
+			return false, errors.Errorf("error getting operator deployment: %s", err)
+		}
```

Comment:
I have one concern here, since we call the wait function which performs a Get operation immediately after we create the Deployment, it could be possible that we receive a not found error here. When an error is returned in `wait.Poll` it immediately exits with the error. We could maybe try to simply return `false` in the case of an `NotFound` error. @irajdeep @FlorentinaSimlinger what do you think?

Reply:
Sure, to be extra cautious we can check for `NotFound` and return nil here.